packages/client-cli/cli.mjs

```diff
@@ -48,60 +52,111 @@ async function writeGraphQLClient (folder, name, schema, url) {
-          'content-type': 'application/json'
-        },
-        body: JSON.stringify({
-          query
-        })
-      })
+      /* c8 ignore next 3 */
+    } catch (err) {
+      return false
+    }
+    return true
+  }
+  res.body.resume()
 
-      const text = await res.body.text()
+  return false
+}
 
-      if (res.statusCode !== 200) {
-        throw new Error('Could not download file')
-      }
+async function downloadAndWriteGraphQL (logger, url, folder, name) {
+  logger.info(`Trying to download GraphQL schema from ${url}`)
+  const query = graphql.getIntrospectionQuery()
+  let res
 
-      const { data: schema } = JSON.parse(text)
-      await writeGraphQLClient(folder, name, schema, url)
-    }
+  try {
+    res = await request(url, {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json'
+      },
+      body: JSON.stringify({
+        query
+      })
+    })
   } catch (err) {
+    /* c8 ignore next 6 */
     if (
       err.code !== 'ERR_INVALID_URL' &&
       err.code !== 'UND_ERR_INVALID_ARG'
     ) {
       throw err
     }
+    return false
+  }
 
-    const text = await readFile(url, 'utf8')
+  const text = await res.body.text()
 
-    // try OpenAPI first
-    try {
-      await writeOpenAPIClient(folder, name, text, fullResponse)
-    } catch {
-      // try GraphQL
-      const schema = graphql.buildSchema(text)
-      const introspectionResult = graphql.introspectionFromSchema(schema)
+  if (res.statusCode !== 200) {
+    return false
+  }
 
-      // dummy URL
-      await writeGraphQLClient(folder, name, introspectionResult, 'http://localhost:3042/graphql')
-    }
+  const { data: schema } = JSON.parse(text)
+  await writeGraphQLClient(folder, name, schema, url)
+  return true
+}
+
+async function readFromFileAndWrite (logger, file, folder, name, fullResponse) {
+  logger.info(`Trying to schema from file ${file}`)
+  const text = await readFile(file, 'utf8')
+
+  // try OpenAPI first
+  try {
+    await writeOpenAPIClient(folder, name, text, fullResponse)
+    return true
+  } catch {
+    // try GraphQL
+    const schema = graphql.buildSchema(text)
+    const introspectionResult = graphql.introspectionFromSchema(schema)
+
+    // dummy URL
+    await writeGraphQLClient(folder, name, introspectionResult, 'http://localhost:3042/graphql')
+    return true
+  }
+}
+
+async function downloadAndProcess ({ url, name, folder, config, r: fullResponse, logger, runtime }) {
+  let found = false
+  const toTry = [
+    downloadAndWriteOpenAPI.bind(null, logger, url + '/documentation/json', folder, name, fullResponse),
+    downloadAndWriteGraphQL.bind(null, logger, url + '/graphql', folder, name),
+    downloadAndWriteOpenAPI.bind(null, logger, url, folder, name, fullResponse),
+    downloadAndWriteGraphQL.bind(null, logger, url, folder, name),
+    readFromFileAndWrite.bind(null, logger, url, folder, name, fullResponse)
+  ]
+
+  while (!found) {
+    found = await toTry.shift()()
```

Comment:
Is there any chance that this will crash (shifting until the array is empty and trying to call the function)